exporters/otlp/all/src/main/java/io/opentelemetry/exporter/otlp/http/metrics/OtlpHttpMetricExporterBuilder.java

```diff
@@ -215,6 +215,13 @@ public OtlpHttpMetricExporterBuilder setRetryPolicy(RetryPolicy retryPolicy) {
     return this;
   }
 
+  /** Sets the proxy to be used. */
+  public OtlpHttpMetricExporterBuilder setProxy(String proxyHost, int proxyPort) {
```

Comment:
Hi thanks for the contribution. The crux of this will be determine the appropriate format to expose proxy information. Given that we have strict backwards compatibility guarantees we'll be stuck supporting whatever we come up for the foreseeable future. I want to make sure that the configuration is exhaustive of what users would commonly expect, and plan on verifying this by evaluating how a number of the popular http client libraries expose proxy configuration. This will take some time to get to for me, so not realistic to get it done before the 2/9/24 release, but could be on the table for march.

Reply:
sometimes forward proxies require a username and password, though much less frequently.

maybe something like `setProxy(ProxyOptions)` and `ProxyOptions.builder().setHost(...).setPort(...).build()` to give room for expansion?

Reply:
> maybe something like setProxy(ProxyOptions) and ProxyOptions.builder().setHost(...).setPort(...).build() to give room for expansion?

Yup that's a good thought. If we determine there are say more than 2 parameters, and that the parameters are sometimes present and sometimes not, a build seems appropriate.

Reply:
Okay, I can file a change for this part. @jack-berg Would this give some more traction to the approval process also in terms of the discussion for being backwards compatible?

Reply:
> maybe something like setProxy(ProxyOptions) and ProxyOptions.builder().setHost(...).setPort(...).build() to give room for expansion?
@trask As setting the host and port are essential, I would rather put this as parameter in the build method than to make it some kind of optional having to call setHost and setPort optionally. Or would you rather like to have validation for this inside the builder and still not supply them using the method?
In the end it would look like `ProxyOptions.builder(String host, int port).build()`, still being open for extension.

Reply:
> In the end it would look like `ProxyOptions.builder(String host, int port).build()`, still being open for extension.

Yup, that's right. Builders which have parameters the user is required to provide accept those as initialization arguments for the builder.

Reply:
I have replaced the `proxyHost` and `proxyPort` with a new `ProxyOptions` object, please check.